app/services/CISSessionService.scala

```diff
@@ -16,31 +16,42 @@
 
 package services
 
+import common.SessionValues.TEMP_EMPLOYER_REF
 import connectors.IncomeTaxUserDataConnector
 import models._
 import models.mongo.{CisCYAModel, CisUserData, DataNotUpdatedError, DatabaseError}
 import org.joda.time.DateTimeZone
+import play.api.Logging
+import play.api.mvc.Request
 import repositories.CisUserDataRepository
 import uk.gov.hmrc.http.HeaderCarrier
 import utils.Clock
 
+import java.time.Month
 import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 
 class CISSessionService @Inject()(cisUserDataRepository: CisUserDataRepository,
                                   incomeTaxUserDataConnector: IncomeTaxUserDataConnector,
-                                  clock: Clock)(implicit val ec: ExecutionContext) {
+                                  clock: Clock)(implicit val ec: ExecutionContext) extends Logging {
 
-  def getSessionData(taxYear: Int, employerRef: String, user: User): Future[Either[DatabaseError, Option[CisUserData]]] = {
-    cisUserDataRepository.find(taxYear, employerRef, user)
+  def getSessionData(taxYear: Int, employerRef: String, user: User)
+                    (implicit request: Request[_]): Future[Either[DatabaseError, Option[CisUserData]]] = {
+
+    val tempEmployerRef = request.session.get(TEMP_EMPLOYER_REF)
```

Comment:
This can be passed as a parameter, instead of passing the whole request. This is what we want to avoid doing in services